File: jimmer-example/spring-boot-with-jimmer-multiple-datasource/src/main/java/org/jimmer/jimmerdatasouce/config/SqlClientConfig.java
package org.jimmer.jimmerdatasouce.config;

import org.babyfish.jimmer.spring.SqlClients;
import org.babyfish.jimmer.spring.transaction.JimmerTransactionManager;
import org.babyfish.jimmer.spring.transaction.TransactionalSqlClients;
import org.babyfish.jimmer.sql.JSqlClient;
import org.babyfish.jimmer.sql.runtime.ConnectionManager;
import org.babyfish.jimmer.sql.runtime.DatabaseValidationMode;
import org.babyfish.jimmer.sql.runtime.Executor;
import org.babyfish.jimmer.sql.runtime.SqlFormatter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.transaction.PlatformTransactionManager;

import javax.sql.DataSource;

/**
 * 该类主要配置多数据源事务
 * @author jhlz
 * @version x.x.x
 */
@Configuration
public class SqlClientConfig {

    @Bean("tm1")
    public PlatformTransactionManager tm1(ApplicationContext ctx, @Qualifier("ds1") DataSource dataSource) {
        return new JimmerTransactionManager(SqlClients.java(ctx, dataSource, null));
    }

    @Bean("tm2")
    public PlatformTransactionManager tm2(ApplicationContext ctx, @Qualifier("ds2") DataSource dataSource) {
        return new JimmerTransactionManager(SqlClients.java(ctx, dataSource, null));
    }

    @Bean
    public JSqlClient sqlClient() {
        return TransactionalSqlClients.java();
    }

}

File: jimmer-example/spring-boot-with-jimmer-kt/src/main/kotlin/org/jimmer/config/SqlClientConfig.kt
package org.jimmer.config

import org.babyfish.jimmer.spring.SqlClients
import org.babyfish.jimmer.spring.transaction.JimmerTransactionManager
import org.babyfish.jimmer.spring.transaction.TransactionalSqlClients
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.transaction.PlatformTransactionManager
import javax.sql.DataSource

/**
 * jimmer 多数据源配置
 * @author jhlz
 * @version x.x.x
 */
@Configuration
class SqlClientConfig {
    @Bean
    fun tm1(
        ctx: ApplicationContext,
        @Qualifier("ds1") dataSource: DataSource
    ): PlatformTransactionManager =
        JimmerTransactionManager(
            SqlClients.kotlin(ctx, dataSource, null)
        )

    @Bean
    fun tm2(
        ctx: ApplicationContext,
        @Qualifier("ds2") dataSource: DataSource
    ): PlatformTransactionManager =
        JimmerTransactionManager(
            SqlClients.kotlin(ctx, dataSource, null)
        )

    @Bean
    fun sqlClient(): KSqlClient = TransactionalSqlClients.kotlin()
}